apiserver/facades/client/modelmanager/modelmanager.go

```diff
@@ -806,55 +806,67 @@ func (m *ModelManagerAPI) ListModelSummaries(ctx context.Context, req params.Mod
 		return result, errors.Trace(err)
 	}
 
-	modelInfos, err := m.state.ModelSummariesForUser(userTag, req.All && m.isAdmin)
+	var modelInfos []model.ModelSummary
+	if req.All {
+		if !m.isAdmin {
+			return result, fmt.Errorf("%w: cannot list all models as non-admin user", apiservererrors.ErrPerm)
+		}
+		modelInfos, err = m.modelService.ListAllModelSummaries(ctx)
+	} else {
+		modelInfos, err = m.modelService.ListModelSummariesForUser(ctx, userTag.Id())
+	}
 	if err != nil {
 		return result, errors.Trace(err)
 	}
+
 	err = m.fillInStatusBasedOnCloudCredentialValidity(ctx, modelInfos)
 	if err != nil {
-		return result, errors.Trace(err)
+		return result, apiservererrors.ServerError(
+			errors.Annotatef(err, "listing model summaries: filling in status for missing cloud credential"),
+		)
 	}
 
 	for _, mi := range modelInfos {
```

Comment:
I can't see where `result.Results` is being initialised. We can pre allocate some memory as a quick optimisation as well for this.

Reply:
Changed, and removed the append to replace with an allocation to [i]